perses/tests/test_relative.py

```diff
@@ -1047,3 +1047,112 @@ def test_RESTCapableHybridTopologyFactory_energies_GPU():
+        chain_id = '1'
+        residue_id = '42'
+        proposed_residue = 'ALA'
+        solvent_delivery = PointMutationExecutor(input_filename,
+                                                 chain_id,
+                                                 residue_id,
+                                                 proposed_residue,
+                                                 forcefield_files=['amber14/protein.ff14SB.xml', 'amber14/tip3p.xml'],
+                                                 ionic_strength=0.05 * unit.molar,
+                                                 rest_radius=0.2,
+                                                 generate_unmodified_hybrid_topology_factory=True,
+                                                 generate_rest_capable_hybrid_topology_factory=True,
+                                                 conduct_endstate_validation=False,
+                                                 )
+        htf = solvent_delivery.get_apo_rest_htf()
+        vanilla_htf = solvent_delivery.get_apo_htf()
+        htfs = [htf, vanilla_htf]
+
+    for htf in htfs:
+        # Modify the htf for tests
+        # For these tests, we need to turn the LRC on for the CustomNonbondedForce, since the LRC is on for the real systems
+        force_dict = {force.getName(): index for index, force in enumerate(htf.hybrid_system.getForces())}
+        if htf.__class__.__name__ == 'HybridTopologyFactory':
+            htf.hybrid_system.getForce(force_dict['CustomNonbondedForce']).setUseLongRangeCorrection(True)
+            htf.hybrid_system.getForce(9).setName("CustomBondForce_exceptions")  # TODO: This can be removed once PR #1022 is merged
+        elif htf.__class__.__name__ == 'RESTCapableHybridTopologyFactory':
+            htf.hybrid_system.getForce(force_dict['CustomNonbondedForce_sterics']).setUseLongRangeCorrection(True)
+
+        # Generate unsampled endstates
+        unsampled_endstates = create_endstates_from_real_systems(htf, for_testing=True)
+
+        # Check to make sure energies are the same in the unsampled endstate hybrid system as they are in the original hybrid system
+        for endstate in [0, 1]:
+            if use_point_energies:
+                validate_unsampled_endstates_point(htf, unsampled_endstates[endstate].system, endstate, minimize=True)
+            if use_md_energies:
+                validate_unsampled_endstates_md(htf, unsampled_endstates[endstate].system, endstate, n_steps=10, save_freq=1)
+
+def test_unsampled_endstate_energies():
+    """
+    Uses run_unsampled_endstate_energies() to run energy validation for the unsampled endstates generated for
+    RESTCapableHybridTopologyFactory and HybridTopologyFactory.
+
+    Test system: alanine dipeptide in solvent.
+    """
+
+    # Run point energy validation test
+    run_unsampled_endstate_energies('ala-dipeptide', use_point_energies=True)
+
+@pytest.mark.gpu_needed
```

Comment:
@ijpulidos Have you benchmarked how long this takes?

Reply:
He hasn't, but when I ran these in a jupyter notebook with an interactive gpu (probably an lt node) on lilac, the `validate_unsampled_endstates_md()` take 7 minutes for barstar.